### ISSUE TITLE
linux-sunxi-3.4: duplicate kernel messages

### ISSUE THREAD

**Issue by polarcat** [linux-sunxi/linux-sunxi#75]
Kernel messages are duplicated when printed on console.

**Comment by polarcat** [comment]
The patch is coming. The problem appears to be in kernel/printk.c implementation: to workaround the issue I decided to backport printk stuff from 3.6. Please comment.

**Comment by hno** [comment]
That's great

**Comment by amery** [comment]
if you have a simple fix for that, please, go ahead. it's a very annoying problem on both, 3.0 and 3.4 trees.

**Comment by polarcat** [comment]
i attached patch to reply on hno's comment, but mail never appeared here :(

**Comment by amery** [comment]
github suggests using gist and paste a link, or a pull request or an inline patch.... not nice. 

I'm kind of starting to hate this issue tracker.

I created a ML at https://groups.google.com/forum/?nomobile=true#!forum/linux-sunxi ... but not really happy with the idea of @googlegroups.com either.

**Comment by polarcat** [comment]
>    github suggests using gist and paste a link, or a pull request or an
>    inline patch.... not nice.
> 
>    I'm kind of starting to hate this issue tracker.
> 
>    I created a ML at
>    [1]https://groups.google.com/forum/?nomobile=true#!forum/linux-sunxi
>    ... but not really happy with the idea of @googlegroups.com either.

```
what about dev@linux-sunxi.org? see issue #73
```

**Comment by polarcat** [comment]
little experiment: sending pull request as a comment.

The following changes since commit 4eb3265978cc006aa63e0bfda366abe99c57c037:

  video: sunxi: rename CONFIG_SUNXI_FB\* to CONFIG_FB_SUNXI\* (2012-09-06 16:32:35 +0200)

are available in the git repository at:
  git@github.com:polarcat/linux-allwinner.git linux-sunxi-3.4

Aliaksei Katovich (1):
      printk: fix duplicated messages on console

 include/linux/kern_levels.h |   25 +
 include/linux/kmsg_dump.h   |   61 ++-
 include/linux/printk.h      |   52 +-
 kernel/printk.c             | 2009 ++++++++++++++++++++++++++++++++-----------
 4 files changed, 1599 insertions(+), 548 deletions(-)
 create mode 100644 include/linux/kern_levels.h

**Comment by polarcat** [comment]
ignore previous comment. sorry for spam. will start using github's pull-requests as everyone else.

**Comment by amery** [comment]
nah, a link like https://github.com/polarcat/linux-allwinner/commit/f5ece65988483b38e3ec48b88d84720531b04425 is enough :-) ... and appending `.patch` gives you the real commit to pass to `git` directly

**Comment by polarcat** [comment]
actually i have no choice at all but send a link as you suggested. Otherwise, when i push pull-request button and then try to select destination repository, github simply exits with error!

**Comment by amery** [comment]
well. anyone has something to comment about the patch itself? :)

**Comment by hno** [comment]
Only that it can't be right to need to backport this huge amount of changes. Stock kernel do not do these duplicate entries.

**Comment by polarcat** [comment]
>    Only that it can't be right to need to backport this huge amount of
>    changes. Stock kernel do not do these duplicate entries.

```
Agree, it is completely unacceptable under normal circumstances, but otoh,
who wants to fix problems that are already fixed somewhere else? I decided
to not waste much time on this, there are more serious issues we have to
work on. Besides, I believe that the ultimate goal is to get forwardported.

PS. IMHO that is one of the reasons to keep the pace with upstream.
PS.PS. Besides, I mentioned already that this is just a workaround.
```
